### PR TITLE
feat: update rule `no-unsupported-ssr-properties` to reflect SSR best practices @W-14387292

### PR DESCRIPTION
**Pull request by seckardt**

### CODE REVIEW

File: lib/rules/no-unsupported-ssr-properties.js [hunk 1]
```diff
@@ -20,7 +23,6 @@ const disallowedProperties = [
     'ownerDocument',
     'querySelector',
     'querySelectorAll',
-    'removeEventListener',
```
Comment (divmain) [inline lib/rules/no-unsupported-ssr-properties.js:23]: Thanks for these changes. Much appreciated!

File: lib/rule-helpers.js [hunk 2]
```diff
@@ -243,10 +246,17 @@ module.exports.noPropertyAccessDuringSSR = function noPropertyAccessDuringSSR(
...
             }
             if (
+                node.parent.type === 'CallExpression' &&
+                node.parent.optional !== true &&
                 node.object.type === 'ThisExpression' &&
                 node.property.type === 'Identifier' &&
                 forbiddenPropertyNames.includes(node.property.name)
             ) {
+                // Prevents expressions like:
+                // this.addEventListener('click', () => { ... });
```
Comment (divmain) [inline lib/rule-helpers.js:256]: Is this a typo, and should be `dispatchEvent` or something? This should function fine in SSR, where `addEventListener` is a no-op.

**Review by divmain — COMMENTED** [review]
Generally LGTM. I did have a couple of questions, and I'd like an answer to the optional call expression nuance.

File: docs/rules/no-unsupported-ssr-properties.md [hunk 3]
```diff
@@ -29,6 +29,42 @@ Examples of **correct** code for this rule:
 ```js
 import { LightningElement } from 'lwc';
 
+export default class Foo extends LightningElement {
+    connectedCallback() {
+        this.querySelector?.('span')?.foo();
```
[the remark this answers is not in the document]
Comment (seckardt) [reply docs/rules/no-unsupported-ssr-properties.md:34]: @divmain Yeah, looking at the runtime code you pointed me to, it seems like the well-intentioned, but problematic "polyfill" behavior in the runtime implements/provides methods like `querySelector`, yet lets them throw an error when called?! That's pretty unfortunate, as due to that behavior the linting rule to catch all posible issues in that area will have to be much more complex and needs to be more contextual to cover forcing the whole statement to either be guarded or fully be made optional, e.g. `a?.foo?.bar?.()`.

File: lib/rule-helpers.js (hunk 2 above)
Comment (seckardt) [reply lib/rule-helpers.js:256]: It's actually a leftover of a former "incarnation" of the PR. In my first version I added `addEventListener` to the disallowed prop as I thought it was just a miss that it's not there. That was before you pointed me to the runtime code with all the "polyfills" on that end. There I saw that actually `removeEventListener` could be removed from that list. I'll update the comment.

**Pull request merged**